src/main/java/com/example/SpringExample/controller/EmployeeController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.SpringExample.controller;
+
+import com.example.SpringExample.entity.Employee;
+import com.example.SpringExample.service.EmployeeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/employees")
+public class EmployeeController {
+
+    @Autowired
+    private  EmployeeService employeeService;
+
+    public EmployeeController(EmployeeService employeeService) {
+        this.employeeService = employeeService;
+    }
+
+
+    @GetMapping
+    public ResponseEntity<List<Employee>> getAllEmployees(){
+        List<Employee> employees = employeeService.getAllEmployees();
+        return new ResponseEntity<>(employees, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Employee> getEmployeeById(@PathVariable long id){
+        Optional<Employee> employee = employeeService.getEmployeeById(id);
```

Comment:
We never return the entity itself. Create an employee response. Map the properties of employee entity to that response and return the response. Make this change in the code wherever applicable.